### PR TITLE
Chore: Enable Go linter gocritic

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
As part of ongoing effort to harmonize Go linting with grafana-plugin-sdk-go, enable also the gocritic linter and fix issues it points out.

The linter mostly finds needlessly complicated code, and in my opinion it does make the code better/more homogeneous. Interested in feedback however, with regards to other people's POV.

We've decided to delay the merging until v7.1.0, just in case it trips people's PRs up.

### CODE REVIEW

File: pkg/services/alerting/rule.go
Comment (papagian) [inline pkg/services/alerting/rule.go:92]: From the [FindAllString()](https://golang.org/pkg/regexp/#Regexp.FindAllString) docs: `A return value of nil indicates no match.`
so maybe this should check if the `matches` is `nil` instead.

**Review by papagian — APPROVED** [review]
LGTM! I have a small suggestion.

File: pkg/services/alerting/rule.go
Comment (aknuds1) [reply pkg/services/alerting/rule.go:92]: @papagian I think it's actually considered OK to check the length of a nil slice in Go, see [this example](https://tour.golang.org/moretypes/12#:~:text=The%20zero%20value%20of%20a,and%20has%20no%20underlying%20array.&text=fmt.,-Println(%22nil!). 

We can check against `nil` instead, but I don't think there's any requirement calling for it :)

Comment (papagian) [reply pkg/services/alerting/rule.go:92]: yes, let's keep it as is.

File: pkg/api/dtos/models.go
Comment (kylebrandt) [inline pkg/api/dtos/models.go:76]: Put into package init, that way a panic it is caught at package init time. 

and also only need to compile once.

File: pkg/plugins/dashboard_importer.go
Comment (kylebrandt) [inline pkg/plugins/dashboard_importer.go:128]: same here with the regex

File: pkg/services/alerting/notifiers/telegram.go
Comment (kylebrandt) [inline pkg/services/alerting/notifiers/telegram.go:128]: string builder would be better but out of scope

**Review by kylebrandt — APPROVED** [review]

**Comment by kylebrandt** [comment]
Approved, but would be good to fix those regexes before merging to avoid the possible panic.

**Comment by aknuds1** [comment]
Thanks @kylebrandt, will take care of it.

**Review by marefr — APPROVED** [review]
LGTM

**Review by jessabe — COMMENTED** [review]
LGTM but as discussed in our weekly yesterday it might be best to hold off merging this until after 7.1

**Review by ryantxu — APPROVED** [review]

**Pull request merged**
